warehouse/packaging/models.py

```diff
@@ -65,13 +62,18 @@ class Role(db.Model):
         CIText,
         ForeignKey("accounts_user.username", onupdate="CASCADE", ondelete="CASCADE"),
     )
-    package_name = Column(
-        Text, ForeignKey("packages.name", onupdate="CASCADE", ondelete="CASCADE")
+    project_id = Column(
+        ForeignKey("packages.id", onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
     )
 
     user = orm.relationship(User, lazy=False)
     project = orm.relationship("Project", lazy=False)
 
+    @property
+    def package_name(self):
+        return project.name
```

Comment:
This looks just wrong, ``project`` isn't defined here. Presumably you meant ``self.project.name``?